src/base/static/libs/maps/mapboxgl-provider.js

```diff
@@ -55,6 +66,148 @@ const mapboxGLMethods = {
   invalidateSize: function() {
     this._map.resize();
   },
+
+  createRasterTileLayer: function(options) {
+    return {
+      id: options.id,
+      type: "raster",
+      source: {
+        type: "raster",
+        tiles: [options.url],
+      },
+    };
+  },
+
+  createVectorTileLayer: function(options) {
+    this._map.addSource(options.id, {
+      type: "vector",
+      tiles: [options.url],
+    });
+
+    return fetch(options.style_url)
```

Comment:
I think we should abstract network requests into a separate api client, like `VectorTileClient`, then call it like so:

```
const layers = await VectorTileClient.fetch(options.style_url)
return layers.map .... // etc
```